packages/utils/tool-utils/src/snapshotNormalizer.ts

```diff
@@ -110,39 +111,54 @@ function getNormalizedBlobContent(blobContent: string, blobName: string): string
 export function getNormalizedSnapshot(snapshot: ITree, config?: ISnapshotNormalizerConfig): ITree {
     // Merge blobs to normalize in the config with runtime blobs to normalize. The contents of these blobs will be
     // parsed and deep sorted.
-    const blobsToNormalize = [ ...config?.blobsToNormalize ?? [] ];
     const normalizedEntries: ITreeEntry[] = [];
 
     for (const entry of snapshot.entries) {
-        switch (entry.type) {
-            case TreeEntry.Blob: {
-                let contents = entry.value.contents;
-                // If this blob has to be normalized or it's a GC blob, parse and sort the blob contents first.
-                if (blobsToNormalize.includes(entry.path) || entry.path.startsWith(gcBlobPrefix)) {
-                    contents = getNormalizedBlobContent(contents, entry.path);
-                }
-                normalizedEntries.push(new BlobTreeEntry(entry.path, contents));
-                break;
-            }
-            case TreeEntry.Tree: {
-                normalizedEntries.push(new TreeTreeEntry(entry.path, getNormalizedSnapshot(entry.value, config)));
-                break;
-            }
-            case TreeEntry.Attachment: {
-                normalizedEntries.push(new AttachmentTreeEntry(entry.path, (entry.value).id));
-                break;
-            }
-
-            default:
-                throw new Error("Unknown entry type");
-        }
+        normalizedEntries.push(normalizeEntry(entry, config));
     }
 
-    // Sory the tree entries based on their path.
+    // Sort the tree entries based on their path.
     normalizedEntries.sort((a, b) => a.path.localeCompare(b.path));
 
     return {
         entries: normalizedEntries,
         id: snapshot.id,
     };
 }
+
+function normalizeEntry(
+    entry: ITreeEntry,
+    config: ISnapshotNormalizerConfig | undefined,
+): ITreeEntry {
+    switch (entry.type) {
+        case TreeEntry.Blob: {
+            let contents = entry.value.contents;
+            // If this blob has to be normalized or it's a GC blob, parse and sort the blob contents first.
+            if (config?.blobsToNormalize?.includes(entry.path) || entry.path.startsWith(gcBlobPrefix)) {
+                contents = getNormalizedBlobContent(contents, entry.path);
+            }
+            return new BlobTreeEntry(entry.path, contents);
+        }
+        case TreeEntry.Tree: {
+            if(config?.excludedChannelContentTypes !== undefined) {
+                for(const maybeAttributes of entry.value.entries) {
+                    if(maybeAttributes.type === TreeEntry.Blob && maybeAttributes.path === ".attributes") {
```

Comment:
nothing guarantees the structure will match. for instance, merge tree supports a variable number of blobs. it will be consistent for the same content, but in matrix the content can vary. if the permutation vectors were large, we could hit a boundary condition where one has header and body, and the other only has header. so here i kept it simple, and we only keep the attributes.

Reply:
makes sense, thanks.